README.md

```diff
@@ -272,6 +272,38 @@ We use the offical [style guide](https://kotlinlang.org/docs/reference/coding-co
 
 Note: ktlint does not allow group imports using `.*`. You can configure IntelliJ to disallow this by going to preferences `Editor > Code Style > Kotlin > Imports` and select "Use single name imports".
 
+
+## Writing Tests
+
+Currently all unit tests shoul be place in the `test/` project instead of `packages/library`. The reason for this is that we need to apply the Realm Compiler Plugin to the tests and this introduces a circular dependency if the tests are in `library`.
```

Comment:
We could, but I suspect that would be a little confusing. Sometimes it isn't immediately clear which tests require the compiler plugin vs. the ones that do not, e.g. The RealmConfigurationTests are actually modified because the constructor calls are rewired.

Reply:
So it is probably simpler to say that all tests to to `test` for now?

Reply:
👍 Yeah. Forgot we introduced that. It was just to try to find some other separation barrier. Maybe we could just put those tests in a separate package.

Reply:
Which tests?

Reply:
You mean those requiering the compiler plugin vs. those that do not?

Reply:
Pure `unit test`-like API tests as the `RealmConfiguration`-tests that are somehow different from all the model-feature tests. One issue with the java-tests is that it hard to locate tests from some general rule on where to navigate. I was just trying to propose some separation scheme. But maybe it is just me that gets lots over there.

Reply:
Okay, yeah. I'm not sure I see them as totally different though, RealmConfiguration is also part of the API, but yeah, another input to the discussion we had yesterday. We should try to come up with some general rules for how to organize tests, so they are easier to find and expand on.

I do agree that the Realm Java Tests are a nightmare to navigate.